nasa/fprime
# GDS CLI Documentation

## F´ Pull Request
|**Field**|**Value**|
|:---|:---|
|**_Submission Date_**| 7/23/2020|
|**_Submitter_**| JHDeerin|
|**_Originating Project/Creator_**| fprime|
|**_Base Branch_**| devel|
|**_Short Description_**| Adds documentation for the GDS CLI feature|
|**_Effected Component_**|  N/A|
|**_Effected Architectures(s)_**| N/A|
|**_Related Issue(s)_**|  N/A|
|**_Has Unit Tests (y/n)_**|  N/A|
|**_Build Checked (y/n)_**|  N/A|
|**_Unit Tests Run (y/n)_**|  N/A|
|**_Documentation Included (y/n)_**|  y|

---
## Change Description

Adds 2 files to the F' `docs` folder (1st in the general folder, 2nd in `Gds/docs`):

- `GDS_CLI_USER_GUIDE.md`, which users can read to learn how to use the GDS CLI tool for themselves
    - The audience here is expected to be familiar with F' in general and have a working knowledge of the GDS, but no prior experience with the GDS CLI
- `GDS_CLI_DEV_GUIDE.md`, which provides helpful information about how the GDS CLI tool is built for future developers working on it

## Rationale

Proper documentation is critical to teaching users how to use the different parts of F', and can be important time-savers for developers seeking to understand a system if written properly.

## Testing Recommendations

- Read through the documents and make sure they're clear, concise, readable, and have no critical information missing. If possible, have someone unfamiliar with the GDS CLI read through this and get their feedback/confusion.
- Confirm developer documentation is actually useful; note any information you would've liked to see that's currently missing, or anything that seems like cruft
- Make sure documents (especially the User Guide) aren't unprofessional in tone
- Confirm included images are sufficiently compressed and their file sizes acceptable

## Future Work

- Keep these documents updated with major changes to the GDS CLI
- Add troubleshooting tips to the user document as needed
- If necessary, add missing information and remove confusing/unnecessary segments
